### DISCUSSION TITLE
esp32 subscribe Mqtt

### DISCUSSION THREAD

**Discussion by afshinjian** [Hieromon/AutoConnect#420] [How can I?] [answered]
Hi Hieromon
your examples in Mqtt works well .
I want to subscribe ESP32 to some Topics to recieve data from other publishers .
Do you have any idea or examples about this?
Tnx

**Comment by Hieromon** [comment, marked as answer]
An example of MQTT subscription with ESP8266 is available on ThingSpeak.com.
https://jp.mathworks.com/help/thingspeak/mqtt-publish-and-subscribe-with-esp8266.html
You can get a template for MQTT subscribers by modifying the logic related to the WiFi connection in that sketch by following the steps in the [AutoConnect documentation](https://hieromon.github.io/AutoConnect/basicusage.html#embed-to-the-sketches).

**Reply by afshinjian** [reply]
Hi Hieromon 
It was a good tip.
Thank you for being responsive and accurate.
It works well.

**Reply by afshinjian** [reply]
Hello again
Thank you for your help.
This is a code that I changed your mqttRSSI.ino  sample a bit and it works properly.
Of course, I did not use ThingSpeak it and I used my local broker.
Maybe it will be useful for someone.
I'm not as professional as you, and I think it would be a great help to your library enthusiasts if you remodel and refine the code a bit.
Yours sincerely
[mqttRSSISUBTOPIC.txt](https://github.com/Hieromon/AutoConnect/files/7572815/mqttRSSISUBTOPIC.txt)
